tensorboard/webapp/metrics/effects/index.ts

```diff
@@ -199,9 +199,13 @@ export class MetricsEffects implements OnInitEffects {
      */
     const requests: TimeSeriesRequest[] = fetchInfos.map((fetchInfo) => {
       const {plugin, tag, runId, sample} = fetchInfo;
-      return isSingleRunPlugin(plugin)
-        ? {plugin, tag, sample, runId: runId!}
-        : {plugin, tag, sample, experimentIds};
+      const partialRequest: TimeSeriesRequest = isSingleRunPlugin(plugin)
+        ? {plugin, tag, runId: runId!}
+        : {plugin, tag, experimentIds};
+      if (sample !== undefined) {
```

Comment:
It seems you have two major choices for fixing errors arising from this new compiler option:
1. Convert property from `property?: type` to `property: type | undefined`
2. Keep `property?: type` but guard setting the property only if we know there is a value to set it to -- for example: `if (valueToSet !== undefined) property.type = valueToSet`

It seems like you applied these options roughly equally in this change. Did you have some methodology to deciding whether to use option 1 or option 2?

Reply:
Binary classification roughly happened like this:

1. is this an internal type where we can be more explicit about existence of a property? If so, I preferred `<T | undefined>` over `prop?: T` since conditionally forming a data structure that conditionally has a property is more difficult and is harder to validate. For instance, `foo.hasOwnProperty('propName')` forces `"propName"` to be understood by the compiler and mangle accordingly or becomes incompatible with mangler like Closure compiler. 
2. if server responses/requests can omit a property, requires a property to be omitted, or is more efficient with a property removed than having undefined value for each one of them, I preferred `<T | undefined>`. 

Metrics data source does not do much transformation to convert a backend type to an internal type so this module, specifically, has a lot of the mixed pattern.